app/jobs/shipit/reap_dead_tasks_job.rb

```diff
@@ -7,13 +7,33 @@ class ReapDeadTasksJob < BackgroundJob
     queue_as :default
 
     def perform
-      zombie_tasks = Task.where(status: ZOMBIE_STATES).reject(&:alive?)
-
       Rails.logger.info("Reaping #{zombie_tasks.size} running tasks.")
       zombie_tasks.each do |task|
         Rails.logger.info("Reaping task #{task.id}: #{task.title}")
         task.report_dead!
       end
     end
+
+    MINUTES_TO_CONSIDER_TASK_RECENT = 5
+    private_constant :MINUTES_TO_CONSIDER_TASK_RECENT
+    def self.recently_created_at
+      MINUTES_TO_CONSIDER_TASK_RECENT.minutes.ago
+    end
+
+    private
+
+    def zombie_tasks
+      @zombie_tasks ||= Task
+                        .where(status: ZOMBIE_STATES)
+                        .where(
```

Comment:
I think this logic should live in a method called `Task.zombie`.

Reply:
Addressed by: https://github.com/Shopify/shipit-engine/pull/1055/commits/45ac417579572572b9c6bb1fe85921871bca2921